libmscore/edit.cpp

```diff
@@ -1369,8 +1369,12 @@ void Score::cmdAddOttava(OttavaType type)
 void Score::cmdSetBeamMode(Beam::Mode mode)
       {
       for (ChordRest* cr : getSelectedChordRests()) {
-            if (cr)
-                  cr->undoChangeProperty(Pid::BEAM_MODE, int(mode));
+            if (cr) {
+                  if (cr->isRest() && (mode == Beam::Mode::AUTO))
+                        cr->undoChangeProperty(Pid::BEAM_MODE, int(Beam::Mode::NONE));
```

Comment:
Shouldn't the issue rather be fixed in the code that interprets that `AUTO` beam mode? The current version of this PR just makes it impossible to set `AUTO` beam mode from the user interface rather than fixing the `AUTO` beam mode meaning for rests. The issue will persist if setting beam mode from a plugin, for example, or when importing files with this beam mode already assigned.

Reply:
`Beam::Mode::NONE` is the default in the rest constructor. AUTO does mean NONE for rests, and that is what the PR does. If you want a rest beamed, set it to any of the other options.
The `resetBeam` method also explicitly sets rests' beam mode to NONE (but does it for either a section or the whole score, whereas this method here does it for a selection only, and does nothing if there is no selection).

Reply:
Not sure how to, or whether at all, deal with rests set to AUTO in existing scores, as there it has a different meaning, and changing it to NONE on reading the score would change the look/layout.

Reply:
Maybe we can use `propertyDefault(Pid::BEAM_MODE)` here instead?
But if so, then probably better in `Score::cmdResetBeamMode()` too.
And adjust `Rest::propertyDefault()`...

Reply:
The problem is that this PR doesn't work at all (i.e. doesn't change the behavoir of applying "Auto" to a selection in regards to the beaming of rests) ...  Any idea where I'm going wrong with it?

Reply:
That 2nd commit (only, 1st not needed, just left in for now) actually makes it working.